packages/components/src/mobile/image-viewer/index.native.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { View, PanResponder, Dimensions, ImageBackground, Image } from 'react-native';
+import Modal from 'react-native-modal';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import styles from './styles.scss';
+import { calculateFullscreenImageSize } from './utils';
+
+// Default Image ratio 4:3
+const IMAGE_ASPECT_RATIO = 4 / 3;
+
+class ImageViewer extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			width: undefined,
+			height: undefined,
+		};
+
+		this.onDimensionsChange = this.onDimensionsChange.bind( this );
+	}
+	
+	componentDidMount() {
+		Dimensions.addEventListener("change", this.onDimensionsChange);
+		this.fetchImageSize();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.url !== prevProps.url ) {
+			this.fetchImageSize();
+		}
+	}
+
+    componentWillUnmount() {
+		Dimensions.removeEventListener("change", this.onDimensionsChange);
+	}
+
+	onDimensionsChange(dims) {
+		this.fetchImageSize();
+	}
+
+	fetchImageSize() {
+		this.container = Dimensions.get("window")
+		Image.getSize( this.props.url, ( width, height ) => {
+			this.image = { width, height };
```

Comment:
It doesn't seem `this.image` is used for anything else that `this.calculateSize()` I think we can refactor a bit here.